From 670fcdf73e85f67e0917cdf0cc6ec4f5fcf43392 Mon Sep 17 00:00:00 2001
From: Swamil Jain <s-jain1@ti.com>
Date: Mon, 30 Jun 2025 18:29:50 +0530
Subject: [PATCH] PENDING: arch: arm64: dts: ti: Move the companion and
 secondary-oldi nodes to overlays

The tidss_oldi driver uses the presence of 'ti,companion-oldi' and
'ti,secondary-oldi' properties to determine whether an OLDI panel is
operating in dual-link mode or clone mode. However, this properties are
set as enabled by default in k3-am62p-j772s-common-main.dtsi, causing
the driver to incorrectly assume panel to be always in dual link or
clone mode even if panel is supporting single-link mode.

To fix this, move the 'ti,companion-oldi' and 'ti,secondary-oldi'
properties to the OLDI panel overlays. This ensures that the driver
correctly detects and configures single-link panels without forcing
clone mode or dual-link mode.

Fixes: d723267dbc5f ("PENDING: arm64: dts: ti: k3-am62p-j722s-common-main: Add support for DSS0")
Reviewed-by: Devarsh Thakkar <devarsht@ti.com>
Signed-off-by: Swamil Jain <s-jain1@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi          | 2 --
 .../boot/dts/ti/k3-am62p5-sk-microtips-mf101hie-panel.dtso      | 2 ++
 .../arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf103hie-lcd2.dtso | 2 ++
 .../boot/dts/ti/k3-j722s-evm-microtips-mf101hie-panel.dtso      | 2 ++
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
index 47a772d7cfe6..7a4c0a758496 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
@@ -1165,7 +1165,6 @@ oldi0_dss0: oldi@0 {
 				reg = <0>;
 				clocks = <&k3_clks 186 0>;
 				clock-names = "serial";
-				ti,companion-oldi = <&oldi1_dss0>;
 				ti,oldi-io-ctrl = <&dss_oldi_io_ctrl>;
 				status = "disabled";
 
@@ -1175,7 +1174,6 @@ oldi0_dss0_ports: ports {
 
 			oldi1_dss0: oldi@1 {
 				reg = <1>;
-				ti,secondary-oldi;
 				status = "disabled";
 
 				oldi1_dss0_ports: ports {
diff --git a/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf101hie-panel.dtso b/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf101hie-panel.dtso
index 983280f6a70f..81c740ccb8c2 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf101hie-panel.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf101hie-panel.dtso
@@ -57,10 +57,12 @@ &dss0 {
 
 &oldi0_dss0 {
 	status = "okay";
+	ti,companion-oldi = <&oldi1_dss0>;
 };
 
 &oldi1_dss0 {
 	status = "okay";
+	ti,secondary-oldi;
 };
 
 &oldi0_dss0_ports {
diff --git a/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf103hie-lcd2.dtso b/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf103hie-lcd2.dtso
index b62240477ef5..939a06f59b88 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf103hie-lcd2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62p5-sk-microtips-mf103hie-lcd2.dtso
@@ -57,10 +57,12 @@ &dss0 {
 
 &oldi0_dss0 {
 	status = "okay";
+	ti,companion-oldi = <&oldi1_dss0>;
 };
 
 &oldi1_dss0 {
 	status = "okay";
+	ti,secondary-oldi;
 };
 
 &oldi0_dss0_ports {
diff --git a/arch/arm64/boot/dts/ti/k3-j722s-evm-microtips-mf101hie-panel.dtso b/arch/arm64/boot/dts/ti/k3-j722s-evm-microtips-mf101hie-panel.dtso
index d5c74fac3d91..d37cf0a6692d 100644
--- a/arch/arm64/boot/dts/ti/k3-j722s-evm-microtips-mf101hie-panel.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j722s-evm-microtips-mf101hie-panel.dtso
@@ -55,10 +55,12 @@ &dss0 {
 
 &oldi0_dss0 {
 	status = "okay";
+	ti,companion-oldi = <&oldi1_dss0>;
 };
 
 &oldi1_dss0 {
 	status = "okay";
+	ti,secondary-oldi;
 };
 
 &oldi0_dss0_ports {
-- 
2.50.1

